Tests/Serilog.Exceptions.Test/Reflection/ReflectionInfoExtractorTest.cs

```diff
@@ -0,0 +1,76 @@
+namespace Serilog.Exceptions.Test.Reflection
+{
+    using System.Linq;
+    using Exceptions.Reflection;
+    using FluentAssertions;
+    using Xunit;
+
+    public class ReflectionInfoExtractorTest
+    {
+        private readonly ReflectionInfoExtractor reflectionInfoExtractor = new();
+
+        [Fact]
+        public void GivenObjectWithRedefinedProperty_ShouldDiscardBaseClassProperty()
+        {
+            var testObject = new TestObjectWithRedefinedProperty() { Name = 123 };
+
+            var reflectionInfo = this.reflectionInfoExtractor.GetOrCreateReflectionInfo(typeof(TestObjectWithRedefinedProperty));
+
+            reflectionInfo.Properties.Should().HaveCount(2);
+
+            var namePropertyInfo = reflectionInfo.Properties.Should().ContainSingle(x => x.Name == "Name").Which;
+            namePropertyInfo.Name.Should().Be(nameof(TestObjectWithRedefinedProperty.Name));
+            namePropertyInfo.DeclaringType.Should().Be(typeof(TestObjectWithRedefinedProperty));
+            var nameGetter = namePropertyInfo.Getter;
+            var testObjectName = nameGetter(testObject);
+            testObjectName.Should().BeOfType<int>().Which.Should().Be(123);
+
+            var baseClassPropertyInfo = reflectionInfo
+                .Properties.Should().ContainSingle(x => x.Name == "Serilog.Exceptions.Test.Reflection.ReflectionInfoExtractorTest+TestObject.Name").Which;
```

Comment:
Do you think we can get away with just using the type name instead of the full namespace too? So `TestObject.Name`? The namespace can get quite verbose for logs.

Reply:
We can, and I guess that makes sense especially that we have our last line of defense in `ExceptionPropertiesBag`.
However, that would be a breaking change, because in one case `ReflectionBasedDestructurer` was detecting duplicate names and attached `FullName`. 

What would you prefer?

Reply:
> in one case ReflectionBasedDestructurer was detecting duplicate names and attached FullName

When was it doing that?

Reply:
![image](https://user-images.githubusercontent.com/3998666/119334259-2f998b80-bc8b-11eb-9859-86b33c763713.png)
You can find it in the PR diff.

Reply:
I have replaced it with just using the name since this matter is not handled elsewhere.
I found it only because there was a test written which failed when I tried to use just type name.
My decision was made only because I wanted to avoid breaking changes.
I would prefer using just type name instead of full name, is you ask me.

Reply:
I agree, if we explicitly warn about that change of behavior, it would be fine.
I will adjust.

Reply:
Done.